### PR TITLE
feat: move allowance modal to alert

### LINKED ISSUE #2694: [ENG-54] Move allowance modal to alert
Image

<sub>[ENG-54](https://linear.app/heyxyz/issue/ENG-54/move-allowance-modal-to-alert)</sub>

### PR DESCRIPTION
**Pull request by foolo**
## What does this PR do?

Fixes #2694
Fixes ENG-54

https://github.com/lensterxyz/lenster/assets/667227/b72d6fdf-8515-47d3-8ff8-2a4def496ad1

### <samp>🤖 Generated by Copilot at e5f29a8</samp>

Refactored the `AllowanceButton` component to use the native `Alert` component instead of the custom `Modal` component. Simplified the `Button` component by removing the unused `icon` prop.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at e5f29a8</samp>

* Replace `Modal` component with `Alert` component in `AllowanceButton` component to improve user interaction and accessibility ([link](https://github.com/lensterxyz/lenster/pull/3753/files?diff=unified&w=0#diff-cfc16ec8965a8ee0f8d417864ef8445f94b2dea6959342a0f0db76ab033f0408L1-R5), [link](https://github.com/lensterxyz/lenster/pull/3753/files?diff=unified&w=0#diff-cfc16ec8965a8ee0f8d417864ef8445f94b2dea6959342a0f0db76ab033f0408L108-R131))
* Remove unused `ExclamationIcon` import from `Button.tsx` ([link](https://github.com/lensterxyz/lenster/pull/3753/files?diff=unified&w=0#diff-cfc16ec8965a8ee0f8d417864ef8445f94b2dea6959342a0f0db76ab033f0408L1-R5))
* Remove `icon` prop from `Button` component as it is redundant with `Alert` component ([link](https://github.com/lensterxyz/lenster/pull/3753/files?diff=unified&w=0#diff-cfc16ec8965a8ee0f8d417864ef8445f94b2dea6959342a0f0db76ab033f0408L108-R131))

## Emoji

🔄🚨🎨

### CODE REVIEW

**Comment by bigint** [comment]
We can use yellow color button for this IMO, cuz its not a destructive ops!

**Comment by foolo** [comment]
> We can use yellow color button for this IMO, cuz its not a destructive ops!
@bigint Done
Screenshot_2023-09-12_11-58-47

**Comment by bigint** [comment]
We should add spinner to allow button when tx in progress

File: packages/ui/src/Alert.tsx [hunk 1]
```diff
@@ -70,7 +70,13 @@ export const Alert: FC<AlertProps> = ({
...
                     size="lg"
-                    variant={isDestructive ? 'danger' : 'primary'}
+                    variant={
+                      alertType === 'destructive'
+                        ? 'danger'
+                        : alertType === 'caution'
+                        ? 'warning'
+                        : 'primary'
+                    }
                     disabled={isPerformingAction}
```
Comment (foolo) [inline packages/ui/src/Alert.tsx:80]: add spinner button

**Comment by foolo** [comment]
> We should add spinner to allow button when tx in progress

@bigint Done, not sure how the design should look though, with regards to centering text etc, but here's an attempt

https://github.com/lensterxyz/lenster/assets/667227/1b0eee9e-0552-4a6e-945b-c28dfd9d6b07

**Comment by bigint** [comment]
I guess alert will not be suitable for this UI, we can close this and revisit later

**Comment by foolo** [comment]
> I guess alert will not be suitable for this UI, we can close this and revisit later

Ok, sure. I think the current modal looks quite OK actually. Maybe we don't need to change anything, and just close #2694 

with "current modal" I mean this one:
Image

**Comment by foolo** [comment]
> I guess alert will not be suitable for this UI, we can close this and revisit later

@bigint , Ok, sure. I think the current modal looks quite OK actually. Maybe we don't need to change anything, and just close #2694 

with "current modal" I mean this one:
Image

**Pull request closed without merging**